src/basic/socket-util.c

```diff
@@ -235,20 +235,30 @@ int socket_address_parse_and_warn(SocketAddress *a, const char *s) {
 }
 
 int socket_address_parse_netlink(SocketAddress *a, const char *s) {
-        int family;
+        _cleanup_strv_free_ char **v = NULL;
         unsigned group = 0;
-        _cleanup_free_ char *sfamily = NULL;
+        int family, r;
+
         assert(a);
         assert(s);
 
         zero(*a);
         a->type = SOCK_RAW;
 
-        errno = 0;
-        if (sscanf(s, "%ms %u", &sfamily, &group) < 1)
-                return errno > 0 ? -errno : -EINVAL;
+        v = strv_split(s, WHITESPACE);
+        if (!v)
+                return -ENOMEM;
+
+        if (!IN_SET(strv_length(v), 1, 2))
+                return -EINVAL;
+
+        if (strv_length(v) == 2) {
+                r = safe_atou(v[1], &group);
+                if (r < 0)
+                        return r;
+        }
```

Comment:
I think it's a good idea to replace sscanf() with something that has more strict semantics. But instead of strv_split() I'd propose using extract_first_word() to get the first word, and then safe_atou() directly on what remains. That way we don't bother with allocating an strv array that we actually don't really care much about.